### PR TITLE
Ansible configuration parameter for interface, deployments.

### PR DESCRIPTION
**Pull request by jayunit100**
@eparis deployments and interfaces configuration.

### CODE REVIEW

File: ansible/vagrant/Vagrantfile [hunk 1]
```diff
@@ -137,6 +137,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       ansible.groups = groups
       ansible.playbook = "../cluster.yml"
       ansible.limit = "all" #otherwise the metadata wont be there for ipv4?
+      # Example variables which we might set.
+      #		iface: This is for binding flannel to the non-default eth device.
+      #		apiserver_runtime_config: This can be provided to, for example, enable deployments (hpa).
+      # ansible.extra_vars = { flannel_iface: "eth1", apiserver_runtime_config: "extensions/v1beta1/deployments=true" }
```
Comment (eparis) [inline ansible/vagrant/Vagrantfile]: Do you need to just set `flannel_opts` on vbox? (it's not called `flannel_iface` and more)

`apiserver_runtime_config` seems better documented out in ansible (maybe a commented out example in the master/defaults/meta.yaml) and not here in Vagrant stuff.

Should (and unset) flannel_opts be out in the ansible repo?

Comment (jayunit100) [inline ansible/vagrant/Vagrantfile]: yeah thats a good point, we could put it in the params. 
ill do that instead and move it out of Vagrantfile.
will update...

**Comment by jayunit100** [comment]
updated fixed the iface thingy

**Comment by jayunit100** [comment]
ok. tested this on my vbox hack branch,  and its working.

```
Dec 09 08:35:50 kube-master flanneld[5966]: I1209 08:35:50.093785 05966 main.go:264] VXLAN mode exited
Dec 09 08:35:50 kube-master systemd[1]: Stopped Flanneld overlay address etcd agent.
Dec 09 08:35:50 kube-master systemd[1]: Starting Flanneld overlay address etcd agent...
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.107252 06215 main.go:247] Installing signal handlers
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.107570 06215 main.go:205] Using 172.28.128.7 as external interface
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.108992 06215 subnet.go:83] Subnet lease acquired: 172.16.5.0/24
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.112110 06215 main.go:215] VXLAN mode initialized
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.112250 06215 vxlan.go:124] Watching for L2/L3 misses
Dec 09 08:35:50 kube-master flanneld[6215]: I1209 08:35:50.112287 06215 vxlan.go:130] Watching for new subnet leases
Dec 09 08:35:50 kube-master systemd[1]: Started Flanneld overlay address etcd agent.
^[[A^C
[root@kube-master vagrant]# curl localhost:8080/apis/extensions/v1beta1 | grep -B 2 -A 2 depl
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   899  100   899    0     0   5183      0 --:--:-- --:--:-- --:--:--  5196
  "resources": [
    {
      "name": "deployments",
      "namespaced": true
    },
    {
      "name": "deployments/scale",
      "namespaced": true
    },
    {
      "name": "deployments/status",
      "namespaced": true

**Comment by eparis** [comment]
didn't you want to automatically set flannel_opts if you were on vbox?  rather than leave it to the user?

**Comment by jayunit100** [comment]
@eparis I think the important thing is `flannel_opts` and   `apiserver_runtime_config` for this PR.    I think automating interface choice is a broader issue which I can tackle along side the usage of your /etc/hosts thingy , right now all we really know for sure is that it *should* be parameterized.  this is an incremental step in that direction.

**Pull request merged**